### PR TITLE
Add fx4276 [v102] Add unit tests to FeatureFlagManager

### PR DESCRIPTION
**Pull request by electricRGB**
[JIRA reference](https://mozilla-hub.atlassian.net/browse/FXIOS-4276)
#10819 

There is a minor code improvement because of the tests, but it has no impact on v101.

### CODE REVIEW

File: ClientTests/FeatureFlagManagerTests.swift [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+        #else
+            XCTFail("We are in an unknown build type. WHat's happening!? How did we get here?!")
+        #endif
+    }
+
+    func testDefaultNimbusBoolFlags() {
+        // Tests for default settings should be performed on both build and user
+        // prefs separately to ensure that we are getting the expected results on both.
+        // Technically, at this stage, these should be the same.
+        #if MOZ_CHANNEL_RELEASE
```
Comment (lmarceau) [inline ClientTests/FeatureFlagManagerTests.swift:52]: How would that be changed for tests? We are always on the same channel when testing no?

File: ClientTests/FeatureFlagManagerTests.swift [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+    }
+
+    func testDefaultNimbusCustomFlags() {
+        XCTAssertEqual(featureFlags.getCustomState(for: .searchBarPosition), SearchBarPosition.bottom)
+        XCTAssertEqual(featureFlags.getCustomState(for: .startAtHome), StartAtHomeSetting.afterFourHours)
+    }
+
+    // Changing the prefs manually, to make sure settings are respected through
+    // the FFMs interface
+    func testFFMRespectsProfileChangesForBoolSettings() {
```
Comment (lmarceau) [inline ClientTests/FeatureFlagManagerTests.swift:158]: FFM stands for?

Comment (electricRGB) [reply ClientTests/FeatureFlagManagerTests.swift:158]: FeatureFlagsManager

File: ClientTests/FeatureFlagManagerTests.swift (hunk 1 above)
Comment (electricRGB) [reply ClientTests/FeatureFlagManagerTests.swift:52]: Technically always developer, yeah. But I didn't know what's happening on BR. Like if we're building for Beta, are we testing anything? WHat about release? If we don't, then.... yeah, this is probably irrelevant.

Comment (lmarceau) [reply ClientTests/FeatureFlagManagerTests.swift:52]: We don't run unit tests when releasing Beta, Nightly or Release. At least I don't see it in the BR workflows!

File: ClientTests/FeatureFlagManagerTests.swift (hunk 2 above)
Comment (lmarceau) [reply ClientTests/FeatureFlagManagerTests.swift:158]: Just a nit, I'm not a fan of acronyms -> `testManagerRespectsProfileChangesForBoolSettings` ?

Comment (electricRGB) [reply ClientTests/FeatureFlagManagerTests.swift:78]: sure

**Review by lmarceau — APPROVED** [review]

**Comment by electricRGB** [comment]
Screen Shot 2022-05-19 at 2 22 04 PM

**Pull request merged**